ktbridge/src/jsMain/kotlin/app/cash/quickjs/ktbridge/ktBridgeJs.kt

```diff
@@ -15,28 +15,79 @@
  */
 package app.cash.quickjs.ktbridge
 
+import kotlin.reflect.KClass
 import okio.Buffer
 
+/**
+ * Call this to create a bridge into JavaScript from the JVM.
+ *
+ * The KtBridge Kotlin compiler plugin will rewrite calls to this function to insert a third
+ * parameter to invoke the 3-argument overload.
+ */
 fun <T : Any> createBridgeToJs(service: T, jsAdapter: JsAdapter): BridgeToJs<T> =
   JsBridgeToJs(service, jsAdapter)
 
 private class JsBridgeToJs<T : Any>(val service: T, val jsAdapter: JsAdapter) : BridgeToJs<T> {
   override fun invokeJs(funName: String, arguments: ByteArray): ByteArray {
-    val buffer = Buffer()
-    val eachValueBuffer = Buffer()
-    buffer.write(arguments)
-    val argumentCount = buffer.readInt()
-    val arguments = arrayOfNulls<Any?>(argumentCount)
-    for (i in 0 until argumentCount) {
-      val argumentByteCount = buffer.readInt()
-      eachValueBuffer.write(buffer, argumentByteCount.toLong())
-      val argument = jsAdapter.decode(eachValueBuffer, Any::class)
-      arguments[i] = argument
+    error("unexpected call to invokeJs: is KtBridge plugin configured?")
+  }
+}
+
+/**
+ * Don't invoke this manually; instead call the two-argument form and let the compiler rewrite such
+ * calls to use this.
+ *
+ * This is invoked by compiler-plugin-rewritten code.
```

Comment:
Could probably be `@PublishedApi internal fun` then

Reply:
Yeah, that’s much nicer.